public/main.ts

```diff
@@ -132,26 +132,184 @@ const kingdoms = [
+            let cropOption = document.createElement("option");
+            cropOption.id = Crop[crop] + "-crop-option";
+            cropOption.innerHTML = Crop[crop];
+            sellCrops!.appendChild(cropOption);
+
+            stats?.appendChild(grainStat);
+        }
+        else {
+            player.farms.get(crop)!.stockpile += 1;
+        }
+        player.gold-= cropMarket.get(crop)!;
+    }
+
+
+}
+
+// load gui
+function loadGUI() {
+    cropMarket.forEach((value, key, map) => {
+        let button = document.createElement("button");
+        button.innerHTML = Crop[key] + " " + value.toString() + "G";
+        button.onclick = handleBuy;
+        button.id = Crop[key] + "-store";
+        let store = document.getElementById("store-tab");
+        store?.appendChild(button);
+    });
+
+    document.getElementById("gold")!.innerHTML = player.gold.toString();
+
+    player.farms.forEach((value: Farm, key: Crop, map: Map<Crop, Farm>) => {
+        let parent = document.getElementById("gold-grain-stats");
+        let grain = document.createElement("p");
+        grain.id = Crop[key] + "-stats";
+        grain.innerHTML = value.stockpile + " " + Crop[key];
+        parent!.appendChild(grain);
+
+        let farms = document.getElementById("farms");
+        let farmOption = document.createElement("option");
+        farmOption.id = Crop[key] + "-farm-option";
+        farmOption.innerHTML = Crop[key];
+        farms!.appendChild(farmOption);
+
+        let sellCrops = document.getElementById("crops-sale");
+        let cropOption = document.createElement("option");
+        cropOption.id = Crop[key] + "-crop-option";
+        cropOption.innerHTML = Crop[key];
+        sellCrops!.appendChild(cropOption);
+    })
+
+
+
+
+    document.getElementById("people")!.innerHTML = player.idlePopulation.toString();
+    document.getElementById("soldiers")!.innerHTML = player.army.totalSoldiers.toString();
+    document.getElementById("farmers")!.innerHTML = getTotalFarmers().toString();
+}
+function getTotalFarmers() {
+    let totalFarmers = 0;
+    player.farms.forEach((value: Farm, key: Crop, map: Map<Crop, Farm>) => {
+                totalFarmers += value.totalFarmers;
+            })
+    return totalFarmers;
+}
+function updateStats() {
+    document.getElementById("gold")!.innerHTML = player.gold.toString();
+    document.getElementById("soldiers")!.innerHTML = player.army.totalSoldiers.toString();
+    document.getElementById("farmers")!.innerHTML = getTotalFarmers().toString();
+    document.getElementById("people")!.innerHTML = player.idlePopulation.toString();
+    player.farms.forEach((value: Farm, key: Crop, map: Map<Crop, Farm>) => {
+        document.getElementById(Crop[key] + "-stats")!.innerHTML = value.stockpile.toString() + " " + Crop[key];
+        document.getElementById(Crop[key] + "-farm-option")!.innerHTML = Crop[key];
+        document.getElementById(Crop[key] + "-crop-option")!.innerHTML = Crop[key];
+    })
```

Comment:
This could be improved with the following function
```ts
interface Printable { toString(): string }
function setInnerHTML(id: string, content: Printable) {
  document.getElementById(id)!.innerHTML = content.toString();
}
```

Reply:
In fact, this function has uses outside just this function.

Reply:
So there is an issue with the setInnerHTML function.  It does not account for html elements that are not in the DOM yet.  To fix this I changed the original setInnerHTML function to setIdInnerHTML and added a setElementInnerHTML function.

Reply:
also I changed up the semantics of the onload.  I put it inside of the ts code instead of as a function call on body.  Additionally, I added a defer to script and moved it up to header.